pentagon/cli.py

```diff
@@ -65,18 +75,107 @@ def start_project(name, **kwargs):
 
 
 @click.command()
-@click.option('--name', help="Component name", required=True)
-@click.option('--environment', help="Environment")
-@click.option('--cluster', help="Cluster name")
-@click.option('--namespace', help="Namespace")
-@click.option('--version', help="Version")
-def install_component(name, **kwargs):
+@click.pass_context
+@click.argument('component_path')
+@click.option('--data', '-D', multiple=True, help='Individual Key=Value pairs used by the component')
+@click.option('--file', '-f', help='File to read Key=Value pair from (yaml or json are supported)')
+@click.option('--out', '-o', default='./', help="Path to output module result, if any")
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def add(ctx, component_path, additional_args, **kwargs):
+    _run('add', component_path, additional_args, kwargs)
+
+
+@click.command()
+@click.pass_context
+@click.argument('component_path')
+@click.option('--data', '-D', multiple=True, help='Individual Key=Value pairs used by the component')
+@click.option('--file', '-f', help='File to read Key=Value pair from (yaml or json are supported)')
+@click.option('--out', '-o', default='./', help="Path to output module result, if any")
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def get(ctx, component_path, additional_args, **kwargs):
+    _run('get', component_path, additional_args, kwargs)
+
+
+def _run(action, component_path, additional_args, options):
+    logging.basicConfig(level=options.get('log_level'))
+    logging.debug("Importing module Pentagon{}".format(component_path))
+    logging.debug("with options: {}".format(options))
+    logging.debug("and additional arguments: {}".format(additional_args))
+
+    data = {}
     try:
-        pentagon.PentagonComponent(name, kwargs).install()
+        file = options.get('file', None)
+        if file is not None:
+            data = parse_infile(file)
+        data = parse_data(options.get('data', data))
+
     except Exception as e:
+        logging.error("Error parsing data from file or -D arguments")
+        logging.error(e)
+
+    component_class = get_component_class(component_path)
+
+    try:
+        getattr(component_class(data, additional_args), action)(options.get('out'))
+    except TypeError, e:
+        logging.error("No module or class found: {}".format(component_path))
```

Comment:
I'm having a hard time coming up with a scenario where at `TypeError` isn't directly related to a problem finding the module/class. I can update the output and print the traceback if that makes you more comfortable.

Reply:
The errors could come from `.add` or `.get` of the class. Errors could be related to anything that could go wrong in the component itself. 

The `gettattr` gets the "action" then calls it. If the call fails because `component_class` is `None` then the handler for the `TypeError` is accurate. If the class is not `None` (a valid class), then call goes into the "action" method. If anything in that method fails with `TypeError` this will say it couldn't find the module or class, and then a traceback of what the actual error is.

Reply:
what do you suggest?

Reply:
I don't think `get_component_class` should be allowed to return `None` and raise an exception if it can't find a valid class at that point. Alternatively `component_class` can be checked for for `callable(component_class)` as a conditional. I'd lean towards the exception though as there doesn't seem to a valid use case for calling modules incorrectly.

Reply:
I disagree.  I think returning `None` in this case is appropriate based on the behavior of `locate()` which `get_component_class()` is essentially a wrapper for. I do like the `callable(component_class)` idea though and I'd prefer to implement it that way.